runner/app/live/trickle/media.py

```diff
@@ -81,6 +84,38 @@ def decode_runner():
     loop = asyncio.get_running_loop()
     await loop.run_in_executor(None, decode_runner)
 
+def encode_in(task_pipes, task_lock, image_generator, sync_callback, get_metadata, **kwargs):
+    # encode_av has a tendency to crash, so restart as necessary
+    retryCount = 0
+    last_retry_time = time.time()
+    while retryCount < MAX_ENCODER_RETRIES:
+        try:
+            encode_av(image_generator, sync_callback, get_metadata, **kwargs)
+            break  # clean exit
+        except Exception as exc:
+            current_time = time.time()
+            # Reset retry counter if enough time has elapsed
+            if current_time - last_retry_time > ENCODER_RETRY_RESET_SECONDS:
+                logging.info("Resetting encoder retry count")
+                retryCount = 0
+            retryCount += 1
+            last_retry_time = current_time
+            if retryCount < MAX_ENCODER_RETRIES:
+                logging.exception(f"Error in encode_av, retrying {retryCount}/{MAX_ENCODER_RETRIES}", stack_info=True)
+            else:
+                logging.exception("Error in encode_av, maximum retries reached", stack_info=True)
+            # close leftover writer ends of any pipes to prevent hanging
+            pipe_count = 0
+            with task_lock:
+                pipes = list(task_pipes)
+                for p in pipes:
+                    try:
+                        p.close()
+                        pipe_count += 1
+                    except Exception as e:
+                        logging.exception("Error closing pipe on task list", stack_info=True)
+            logging.info(f"Closed pipes - {pipe_count}")
```

Comment:
nit: would be good to see what the total pipe count is in the same line here so that we can easily tell how many pipes failed to close.

Reply:
added in https://github.com/livepeer/ai-runner/pull/454/commits/36af17a2e27b888d2782ad3fb909376aba90f0a3

fwiw "failing to close" is not necessarily indicative of problem in itself; it may just have crashed after ffmpeg closed but before we had a chance to clean it up. If this happens in practice then it might offer clues about _where_ the crash is happening.